aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -62,28 +65,39 @@
 
     // A component which synchronizes data state between the
     // local storage adapter, and a remote API
-    private final Orchestrator orchestrator;
+    private Orchestrator orchestrator;
 
     // Keeps track of whether of not the category is initialized yet
     private final CountDownLatch categoryInitializationsPending;
 
     // Configuration for the plugin.
-    private AWSDataStorePluginConfiguration pluginConfiguration;
+    private final DataStoreConfiguration userProvidedConfiguration;
+    private final AsyncSubject<DataStoreConfiguration> dataStoreConfigurationProvider;
 
+    @SuppressLint("CheckResult")
     private AWSDataStorePlugin(
-            @NonNull ModelSchemaRegistry modelSchemaRegistry,
             @NonNull ModelProvider modelProvider,
-            @NonNull GraphQlBehavior api) {
+            @NonNull ModelSchemaRegistry modelSchemaRegistry,
+            @NonNull GraphQlBehavior api,
+            @Nullable DataStoreConfiguration userProvidedConfiguration) {
         this.sqliteStorageAdapter = SQLiteStorageAdapter.forModels(modelSchemaRegistry, modelProvider);
         this.storageItemChangeConverter = new GsonStorageItemChangeConverter();
         this.categoryInitializationsPending = new CountDownLatch(1);
-        this.orchestrator = new Orchestrator(
-            modelProvider,
-            modelSchemaRegistry,
-            sqliteStorageAdapter,
-            AppSyncClient.via(api),
-            () -> pluginConfiguration.getBaseSyncIntervalMs()
+        dataStoreConfigurationProvider = AsyncSubject.create();
+
+        dataStoreConfigurationProvider.subscribe(initializedConfiguration -> {
+            orchestrator = new Orchestrator(
```

Comment:
I guess there are two ways of looking at this:

1. The current view:
> The orchestrator is an optional component. It's presence is determined by the content of the configuration. If the configuration does not permit remote sync, then don't create an orchestrator.
2. The original view:
> The orchestrator is a data store component, it is constructed at the same time as the plugin. Later, if the configuration determines that sync should be enabled, the orchestrator is started.

Realistically, the only concrete concern is whether or not `orchestrator` is null at the time you call `start`. So long as configuration is definitely done by the time `initialize()` rolls around, you should be good.